docs/Formula-Android.md

```diff
@@ -0,0 +1,380 @@
+## Formula Android
+The Android module provides a declarative API to connect reactive state management to Android Fragments. 
+This module has been designed for gradual adoption. You can use as much or as little of it as you like.
+
+Some of the goals for this module are:
+
+    - Use single RxJava stream to drive the UI.
+    - Separate state management from Android UI lifecycle.
+    - Ability to group multiple fragments into a flow and share state between them.
+    - Type-safe and scoped fragment event handling. (Avoid casting activity to a listener)
+
+## Using Fragments
+This module provides an API to connect state management and view rendering logic to Android 
+fragments. For this example, we will connect `CounterRenderView` and `CounterFormula` from the 
+main getting started [guide](index.md). 
+
+### Define a fragment key
+Fragment key is used an a unique identifier for a `FormulaFragment`.
+```kotlin
+/**
+ * Fragment key has to provide Parcelable implementation because it is passed 
+ * to the fragment as an argument. 
+ *   
+ * Read more about Parcelize: https://kotlinlang.org/docs/tutorials/android-plugin.html
+ */
+@Parcelize
+data class CounterKey(
+    override val tag: String = "counter"
+) : FragmentKey
+```
+
+### Define a feature factory
+Feature factory creates state observable and a view factory for a fragment with a 
+specific `FragmentKey` type. To continue our example, we define a `CounterFeatureFactory` 
+which will handle `CounterKey` fragments.
+```kotlin
+class CounterFeatureFactory : FeatureFactory<Any, CounterKey> {
```

Comment:
Could this `Any` also be `Unit` if we don't have any dependencies?

Reply:
Currently no, but maybe a good idea to add support for that.